runner/src/server/plugins/engine/models/submission/NotifyModel.ts

```diff
@@ -14,6 +14,23 @@ export type NotifyModel = Omit<
   };
 };
 
+const parseListAsNotifyTemplate = (
+  list: List,
+  model: FormModel,
+  state: FormSubmissionState
+) => {
+  return list.items
+    .filter((item) => checkItemIsValid(model, state, item.condition))
+    .map((item) => `* ${item.value}\n`)
+    .join("");
+};
+
+const checkItemIsValid = (
+  model: FormModel,
+  state: FormSubmissionState,
+  conditionName = ""
+) => conditionName === "" || !!model.conditions[conditionName]?.fn(state);
```

Comment:
!! is generally used as a shortcut to determine whether or not something is defined (because undefined is falsy - so `!` casts it to false), but I would just avoid using this whenever there's a weird truthy/falsy rule involved - so avoid using `!!` if there are booleans or numbers (e.g. 0, 1) won't behave as expected.

```.ts
function checkItemIsValid(model, state, conditionName) {
// I would just allow conditionName to be undefined, i.e. not giving it a default value `""`, unless it gives you other problems?

  // if anything in the chain returns undefined, it will return false `?? false`.

 return model.conditions[conditionName]?.fn?.(state) ?? false;

}

```